src/pages/dashboard/ResourceTypeFiltering.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { Dispatch, FC, SetStateAction, useEffect, useState } from 'react';
+import { QueryObject } from '../../types/search.types';
+import { ResourceFeatureTypes } from '../../types/resource.types';
+import { TFunction, useTranslation } from 'react-i18next';
+import styled from 'styled-components';
+import { Checkbox, FormControl, FormControlLabel, FormGroup } from '@material-ui/core';
+import FormLabel from '@material-ui/core/FormLabel';
+import Typography from '@material-ui/core/Typography';
+
+const StyledFormControl: any = styled(FormControl)`
+  padding-top: 2rem;
+`;
+
+interface ResourceTypeListItem {
+  name: string;
+  isSelected: boolean;
+}
+
+const defaultResourceTypes: string[] = [
+  ResourceFeatureTypes.audio,
+  ResourceFeatureTypes.document,
+  ResourceFeatureTypes.image,
+  ResourceFeatureTypes.presentation,
+  ResourceFeatureTypes.simulation,
+  ResourceFeatureTypes.video,
+];
+
+const initialResourceTypeCheckList = (t: TFunction<string>): ResourceTypeListItem[] => {
```

Comment:
er det ikke like greit å ha denne inni kompoenten?

Reply:
Fikset

Reply:
Den bør nok utenfor likevel. Nå ser jeg at det blir warning  ( React Hook useEffect has a missing dependency: 'initialResourceTypeCheckList'. Either include it or remove the dependency array )  - og hvis vi tar den til følge får vi en evig løkke.

Reply:
Flyttet ut igjen